did_peer/src/numalgos/numalgo2/generate/helpers.rs

```diff
@@ -0,0 +1,256 @@
+use base64::{engine::general_purpose::STANDARD_NO_PAD, Engine};
+use did_doc::schema::{did_doc::DidDocument, verification_method::VerificationMethodKind};
+use did_doc_sov::extra_fields::ExtraFieldsSov;
+
+use crate::{
+    error::DidPeerError,
+    key::get_key_by_verification_method,
+    numalgos::numalgo2::{purpose::ElementPurpose, service_abbreviated::ServiceAbbreviated},
+};
+
+pub fn append_encoded_key_segments(
```

Comment:
`pub(crate)` should suffice

Reply:
especially since this is a bit dangerous api - did is being mutated, but function can fail, leaving `did` in inconsistent state. But I think it's reasonable optimization as far as these functions stay local to the `generate` module

Reply:
Again, this module is not visible outside numalgo2.